### ISSUE TITLE
Some doubts

### ISSUE THREAD

**Issue by QianFuv** [Epidra/CasinoCraft#9]
I translated the mod's language files 👇
1.19.2 forge newest version
zh_cn.json👇
```
{
  "block.casinocraft.cardtable_base_white": "白色卡牌桌",
  "block.casinocraft.cardtable_base_orange": "橙色卡牌桌",
  "block.casinocraft.cardtable_base_magenta": "洋红色卡牌桌",
  "block.casinocraft.cardtable_base_light_blue": "淡蓝色卡牌桌",
  "block.casinocraft.cardtable_base_yellow": "黄色卡牌桌",
  "block.casinocraft.cardtable_base_lime": "酸橙色卡牌桌",
  "block.casinocraft.cardtable_base_pink": "粉红色卡牌桌",
  "block.casinocraft.cardtable_base_gray": "灰色卡牌桌",
  "block.casinocraft.cardtable_base_light_gray": "浅灰色卡牌桌",
  "block.casinocraft.cardtable_base_cyan": "青色卡牌桌",
  "block.casinocraft.cardtable_base_purple": "紫色卡牌桌",
  "block.casinocraft.cardtable_base_blue": "蓝色卡牌桌",
  "block.casinocraft.cardtable_base_brown": "棕色卡牌桌",
  "block.casinocraft.cardtable_base_green": "绿色卡牌桌",
  "block.casinocraft.cardtable_base_red": "红色卡牌桌",
  "block.casinocraft.cardtable_base_black": "黑色卡牌桌",
  "block.casinocraft.cardtable_wide_white": "白色双人卡牌桌",
  "block.casinocraft.cardtable_wide_orange": "橙色双人卡牌桌",
  "block.casinocraft.cardtable_wide_magenta": "洋红色双人卡牌桌",
  "block.casinocraft.cardtable_wide_light_blue": "淡蓝色双人卡牌桌",
  "block.casinocraft.cardtable_wide_yellow": "黄色双人卡牌桌",
  "block.casinocraft.cardtable_wide_lime": "酸橙色双人卡牌桌",
  "block.casinocraft.cardtable_wide_pink": "粉红色双人卡牌桌",
  "block.casinocraft.cardtable_wide_gray": "灰色双人卡牌桌",
  "block.casinocraft.cardtable_wide_light_gray": "浅灰色双人卡牌桌",
  "block.casinocraft.cardtable_wide_cyan": "青色双人卡牌桌",
  "block.casinocraft.cardtable_wide_purple": "紫色双人卡牌桌",
  "block.casinocraft.cardtable_wide_blue": "蓝色双人卡牌桌",
  "block.casinocraft.cardtable_wide_brown": "棕色双人卡牌桌",
  "block.casinocraft.cardtable_wide_green": "绿色双人卡牌桌",
  "block.casinocraft.cardtable_wide_red": "红色双人卡牌桌",
  "block.casinocraft.cardtable_wide_black": "黑色双人卡牌桌",

  "block.casinocraft.arcade_base_white": "白色街机柜",
  "block.casinocraft.arcade_base_orange": "橙色街机柜",
  "block.casinocraft.arcade_base_magenta": "洋红色街机柜",
  "block.casinocraft.arcade_base_light_blue": "淡蓝色街机柜",
  "block.casinocraft.arcade_base_yellow": "黄色街机柜",
  "block.casinocraft.arcade_base_lime": "酸橙色街机柜",
  "block.casinocraft.arcade_base_pink": "粉红色街机柜",
  "block.casinocraft.arcade_base_gray": "灰色街机柜",
  "block.casinocraft.arcade_base_light_gray": "浅灰色街机柜",
  "block.casinocraft.arcade_base_cyan": "青色街机柜",
  "block.casinocraft.arcade_base_purple": "紫色街机柜",
  "block.casinocraft.arcade_base_blue": "蓝色街机柜",
  "block.casinocraft.arcade_base_brown": "棕色街机柜",
  "block.casinocraft.arcade_base_green": "绿色街机柜",
  "block.casinocraft.arcade_base_red": "红色街机柜",
  "block.casinocraft.arcade_base_black": "黑色街机柜",
  "block.casinocraft.arcade_slot_white": "白色老虎机",
  "block.casinocraft.arcade_slot_orange": "橙色老虎机",
  "block.casinocraft.arcade_slot_magenta": "洋红色老虎机",
  "block.casinocraft.arcade_slot_light_blue": "淡蓝色老虎机",
  "block.casinocraft.arcade_slot_yellow": "黄色老虎机",
  "block.casinocraft.arcade_slot_lime": "酸橙色老虎机",
  "block.casinocraft.arcade_slot_pink": "粉红色老虎机",
  "block.casinocraft.arcade_slot_gray": "灰色老虎机",
  "block.casinocraft.arcade_slot_light_gray": "浅灰色老虎机",
  "block.casinocraft.arcade_slot_cyan": "青色老虎机",
  "block.casinocraft.arcade_slot_purple": "紫色老虎机",
  "block.casinocraft.arcade_slot_blue": "蓝色老虎机",
  "block.casinocraft.arcade_slot_brown": "棕色老虎机",
  "block.casinocraft.arcade_slot_green": "绿色老虎机",
  "block.casinocraft.arcade_slot_red": "红色老虎机",
  "block.casinocraft.arcade_slot_black": "黑色老虎机",

  "item.casinocraft.rulebook_1_1": "轮盘",
  "item.casinocraft.rulebook_2_1": "黑杰克",
  "item.casinocraft.rulebook_2_2": "扑克",
  "item.casinocraft.rulebook_3_1": "纸牌接龙",
  "item.casinocraft.rulebook_3_2": "金字塔",
  "item.casinocraft.rulebook_3_3": "Mau Mau",
  "item.casinocraft.rulebook_4_1": "扫雷",
  "item.casinocraft.rulebook_4_2": "石头棋",
  "item.casinocraft.rulebook_5_1": "俄罗斯方块",
  "item.casinocraft.rulebook_5_2": "2048",
  "item.casinocraft.rulebook_6_1": "贪吃蛇",
  "item.casinocraft.rulebook_6_2": "推箱子",
  "item.casinocraft.rulebook_7_0": "老虎机",

  "casinocraft:subtitles.card.place": "卡牌移动",
  "casinocraft:subtitles.card.shove": "卡牌移动",
  "casinocraft:subtitles.chip": "筹码移动",
  "casinocraft:subtitles.pickup": "街机声音",
  "casinocraft:subtitles.dice": "骰子滚动",
  "casinocraft:subtitles.impact": "街机声音",
  "casinocraft:subtitles.menu": "街机声音",
  "casinocraft:subtitles.pause.off": "老虎机声音",
  "casinocraft:subtitles.pause.on": "老虎机声音",
  "casinocraft:subtitles.reward": "有人兑换奖励",
  "casinocraft:subtitles.roulette": "轮盘转动",
  "casinocraft:subtitles.tetris": "街机声音",

  "rule.casinocraft.setting_0": "无限代币存储",
  "rule.casinocraft.setting_1": "无限奖励存储",
  "rule.casinocraft.setting_2": "破坏时掉落物品",
  "rule.casinocraft.setting_3": "不可破坏的方块",
  "rule.casinocraft.setting_4": "每个用户仅一次得分记录",

  "rule.casinocraft.game_11_0_0": "轮盘",

  "rule.casinocraft.game_21_0_0": "黑杰克",

  "rule.casinocraft.game_22_0_0": "扑克",

  "rule.casinocraft.game_31_0_0": "纸牌接龙",
  "rule.casinocraft.game_31_1_0": "1 种花色",
  "rule.casinocraft.game_31_1_1": "2 种花色",
  "rule.casinocraft.game_31_1_2": "3 种花色",
  "rule.casinocraft.game_31_1_3": "4 种花色",
  "rule.casinocraft.game_31_2_0": "升序排列",
  "rule.casinocraft.game_31_2_1": "降序排列",
  "rule.casinocraft.game_31_2_2": "均匀排列",
  "rule.casinocraft.game_31_3_0": "无保留牌",
  "rule.casinocraft.game_31_3_1": "自由细胞",
  "rule.casinocraft.game_31_3_2": "抽取1张牌（覆盖）",
  "rule.casinocraft.game_31_4_0": "放置在任何颜色上",
  "rule.casinocraft.game_31_4_1": "放置在相同颜色上",
  "rule.casinocraft.game_31_4_2": "放置在不同颜色上",

  "rule.casinocraft.game_32_0_0": "寻找匹配的牌",
  "rule.casinocraft.game_32_0_1": "寻找下一张牌",
  "rule.casinocraft.game_32_1_0": "1 座金字塔",
  "rule.casinocraft.game_32_1_1": "2 座金字塔",
  "rule.casinocraft.game_32_1_2": "3 座金字塔",
  "rule.casinocraft.game_32_2_0": "1 条命",
  "rule.casinocraft.game_32_2_1": "2 条命",
  "rule.casinocraft.game_32_2_2": "3 条命",
  "rule.casinocraft.game_32_2_3": "4 条命",
  "rule.casinocraft.game_32_2_4": "5 条命",

  "rule.casinocraft.game_33_0_0": "MauMau",

  "rule.casinocraft.game_41_0_0": "扫雷",

  "rule.casinocraft.game_42_0_0": "石头棋",

  "rule.casinocraft.game_51_0_0": "俄罗斯方块",
  "rule.casinocraft.game_51_0_1": "纵列方块",
  "rule.casinocraft.game_51_0_2": "噗哟噗哟",
  "rule.casinocraft.game_51_1_0": "大场地",
  "rule.casinocraft.game_51_1_1": "小场地",
  "rule.casinocraft.game_51_2_0": "2 块",
  "rule.casinocraft.game_51_2_1": "3 块",
  "rule.casinocraft.game_51_2_2": "4 块",
  "rule.casinocraft.game_51_3_0": "保留按钮",
  "rule.casinocraft.game_51_3_1": "---",

  "rule.casinocraft.game_52_0_0": "2048",
  "rule.casinocraft.game_52_0_1": "神秘方块",
  "rule.casinocraft.game_52_1_0": "撤销按钮",
  "rule.casinocraft.game_52_1_1": "---",

  "rule.casinocraft.game_61_0_0": "贪吃蛇",

  "rule.casinocraft.game_62_0_0": "推箱子",

  "rule.casinocraft.color_0": "红色",
  "rule.casinocraft.color_1": "蓝色",
  "rule.casinocraft.color_2": "黄色",
  "rule.casinocraft.color_3": "绿色",
  "rule.casinocraft.color_4": "橙色",
  "rule.casinocraft.color_5": "紫色",

  "book.casinocraft.head_11_0": "内容",
  "book.casinocraft.page_11_0": "1. 游戏玩法",
  "book.casinocraft.head_11_1": "游戏玩法",
  "book.casinocraft.page_11_1": "你开始时在棋盘上放置一个代币并点击放置确认。如果小游戏启用赌博并且你有足够的物品用于另一个代币，你可以点击再来一个放置另一个代币，否则点击等待让下一个玩家放置他们的代币。每个玩家完成放置后，轮盘将出现并开始旋转。球停下的数字将是获胜号码。获胜号码上的代币将保持颜色并支付奖励，否则它们将变灰。",

  "book.casinocraft.head_21_0": "内容",
  "book.casinocraft.page_21_0": "1. 游戏玩法",
  "book.casinocraft.head_21_1": "游戏玩法",
  "book.casinocraft.page_21_1": "你开始手里有2张牌。你的目标是抽更多的牌，直到你的牌值接近21。超过则输。Ace可以算1或11，视对你更有利。点击双倍下注加倍，放置额外赌注，只拿一张新牌并结束回合。如果起始的2张牌值相同，你可以点击分牌创建第二手牌并放置额外赌注，两手牌开始时各有2张牌。一旦你完成回合，庄家将开始抽牌。庄家抽牌，直到比你大或超过16。庄家比你高且不超过21时，你就输了。",

  "book.casinocraft.head_22_0": "内容",
  "book.casinocraft.page_22_0": "1. 游戏玩法",
  "book.casinocraft.head_22_1": "游戏玩法",
  "book.casinocraft.page_22_1": "你开始手里有5张牌。你的目标是交换牌，尽量获得获胜手牌。你可以用保持按钮设定要保留的牌，点击抽牌后其他牌将被替换。下一步你的牌将重新排列以创建获胜手牌。奖励取决于手牌的好坏。",

  "book.casinocraft.head_31_0": "内容",
  "book.casinocraft.page_31_0": "1. 游戏玩法/b/b2. 规则/b2.1. 花色/b2.2. 排序/b2.3. 保留牌/b2.4. 放置",
  "book.casinocraft.head_31_1": "游戏玩法",
  "book.casinocraft.page_31_1": "目标是将场上的牌放置在右上角的四个完成堆中。",
  "book.casinocraft.head_31_2": "规则 - 花色",
  "book.casinocraft.page_31_2": "此规则决定小游戏开始时放置的不同花色数量。每种花色会增加13张新牌到场上。",
  "book.casinocraft.head_31_3": "规则 - 排序",
  "book.casinocraft.page_31_3": "此规则会改变开始时牌的排列方式。",
  "book.casinocraft.head_31_4": "规则 - 保留牌",
  "book.casinocraft.page_31_4": "无保留牌时，所有牌在开始时都会放置在场上。/b/b使用自由细胞时，所有牌仍会放置在场上，但你可以将牌自由移动到左上角的额外4个格子中。你可以随意取放任何牌。/b/b使用抽取1张牌时，开始时将保留一些牌并放置在保留堆中。点击保留堆可以显示另一张牌，然后可以将其放置在完成堆或场上。",
  "book.casinocraft.head_31_5": "规则 - 放置",
  "book.casinocraft.page_31_5": "决定玩家如何在场上放置牌。",

  "book.casinocraft.head_32_0": "内容",
  "book.casinocraft.page_32_0": "1. 游戏玩法/b/b2. 游戏模式/b2.1. 模式A/b2.2. 模式B/b/b3. 规则/b3.1. 金字塔/b3.2. 生命值",
  "book.casinocraft.head_32_1": "游戏玩法",
  "book.casinocraft.page_32_1": "目标是清除场上的牌。具体方式取决于激活的游戏模式。",
  "book.casinocraft.head_32_2": "游戏模式A",
  "book.casinocraft.page_32_2": "你选择两张牌进行比较。当它们的总和正好是13时，这些牌将从场上移除。保留堆上有2张额外的牌以支持你在场上的导航。可以用箭头按钮循环浏览保留堆。",
  "book.casinocraft.head_32_3": "游戏模式B",
  "book.casinocraft.page_32_3": "你选择场上一张比底部展示牌高1或低1的牌。对于这个小游戏来说，Ace比国王高1。如果你不能再匹配任何牌，点击箭头按钮选择下一张保留堆中的牌。",
  "book.casinocraft.head_32_4": "规则 - 金字塔",
  "book.casinocraft.page_32_4": "这个规则决定了场上的布局。卡牌桌的大小也会影响布局。",
  "book.casinocraft.head_32_5": "规则 - 生命值",
  "book.casinocraft.page_32_5": "这个规则决定玩家有多少条命。当保留堆中的所有牌都被使用并且点击箭头按钮重置堆时，会消耗一条命。失去所有生命时，小游戏结束。",

  "book.casinocraft.head_33_0": "内容",
  "book.casinocraft.page_33_0": "1. 游戏玩法",
  "book.casinocraft.head_33_1": "游戏玩法",
  "book.casinocraft.page_33_1": "这个小游戏通过从手牌中放置一张与中间牌堆上的牌相同花色或点数的牌来进行。如果你不能放置牌，你必须抽一张牌。当一个玩家手中没有牌时，小游戏结束。",

  "book.casinocraft.head_41_0": "内容",
  "book.casinocraft.page_41_0": "1. 游戏玩法",
  "book.casinocraft.head_41_1": "游戏玩法",
  "book.casinocraft.page_41_1": "一定数量的炸弹被放置在场上，然后被灰色石头覆盖。你的目标是点击石头揭开所有的空白格子。揭开一颗炸弹游戏就结束了。石头下的数字会告诉你特定格子周围有多少炸弹。右键点击可以标记一块石头为旗帜。它们不会产生任何效果，但可以帮助玩家记住可能藏有炸弹的地方。/b/b当场上只剩下炸弹时，玩家可以选择再来一轮，使场上有更多炸弹，或者结束游戏。/b/b揭开炸弹输掉游戏会使你的得分减半。",

  "book.casinocraft.head_42_0": "内容",
  "book.casinocraft.page_42_0": "1. 游戏玩法",
  "book.casinocraft.head_42_1": "游戏玩法",
  "book.casinocraft.page_42_1": "在场上的顶部展示一个石头。你的目标是将石头放置在场上标记的位置。小游戏在没有更多位置可以放置石头时结束。/b/b石头只能放置在与另一块石头共享颜色或符号的位置旁边。放置在两个石头旁边时，它必须与第一个石头共享颜色，与第二个石头共享符号。放置在三个石头旁边时，它必须与其他石头共享1个颜色和2个符号或2个颜色和1个符号。放置在四个石头中间时，它必须与其中两个石头共享符号，与另两个石头共享颜色。",

  "book.casinocraft.head_51_0": "内容",
  "book.casinocraft.page_51_0": "1. 游戏玩法/b/b2. 游戏模式/b2.1. 模式A/b2.2. 模式B/b2.3. 模式C/b/b3. 规则/b3.1. 网格大小/b3.2. 方块数量/b3.3. 保留按钮",
  "book.casinocraft.head_51_1": "游戏玩法",
  "book.casinocraft.page_51_1": "不同形状的方块会从网格顶部降下。你可以用方向键移动它们左右方向，并用上键旋转它们。按下键可以让它们立即下落。当没有新的形状可以出现时，游戏结束。",
  "book.casinocraft.head_51_2": "游戏模式A",
  "book.casinocraft.page_51_2": "一旦一整行被填满，整行的方块将被清除。",
  "book.casinocraft.head_51_3": "游戏模式B",
  "book.casinocraft.page_51_3": "当3个或更多相同颜色的方块在水平、垂直或对角线上连成一线时，它们将被清除。这种游戏模式替换了旋转功能，变为颜色循环。",
  "book.casinocraft.head_51_4": "游戏模式C",
  "book.casinocraft.page_51_4": "当4个或更多相同颜色的方块相互连接时，它们将被清除。",
  "book.casinocraft.head_51_5": "规则 - 网格大小",
  "book.casinocraft.page_51_5": "这个规则在大网格和小网格之间切换。",
  "book.casinocraft.head_51_6": "规则 - 方块数量",
  "book.casinocraft.page_51_6": "这个规则决定形状中的方块数量。",
  "book.casinocraft.head_51_7": "规则 - 保留按钮",
  "book.casinocraft.page_51_7": "这个规则激活保留按钮（回车键）。一个形状可以被存放在这里。再次按下时可以与当前形状切换。",

  "book.casinocraft.head_52_0": "内容",
  "book.casinocraft.page_52_0": "1. 游戏玩法/b/b2. 规则/b2.1. 撤销按钮",
  "book.casinocraft.head_52_1": "游戏玩法",
  "book.casinocraft.page_52_1": "网格上有不同数字的方块。按下方向键时，所有方块会朝该方向移动。如果两个相同数字的方块碰撞，它们会合并并形成下一个更高的数字。每次移动后，最低数字的新方块可能会出现在角落。当没有方块可以移动时，游戏结束。",
  "book.casinocraft.head_52_2": "规则 - 撤销按钮",
  "book.casinocraft.page_52_2": "这个规则激活撤销按钮（回车键），允许玩家撤销一次移动。",

  "book.casinocraft.head_61_0": "内容",
  "book.casinocraft.page_61_0": "1. 游戏玩法",
  "book.casinocraft.head_61_1": "游戏玩法",
  "book.casinocraft.page_61_1": "按方向键移动小船到那个方向。当它碰到墙或自己的身体时停下。通过移动到小火花上来获得积分。每次这样做会在你的尾巴上添加一个部分。当你碰到尾巴时，小游戏结束。",

  "book.casinocraft.head_62_0": "内容",
  "book.casinocraft.page_62_0": "1. 游戏玩法",
  "book.casinocraft.head_62_1": "游戏玩法",
  "book.casinocraft.page_62_1": "按方向键将小船移动到那个方向的一个格子。你的任务是将所有的箱子推到标记的位置。但请注意，你只能推，不能拉。当所有箱子都在标记位置时，小游戏结束。",

  "book.casinocraft.head_70_0": "内容",
  "book.casinocraft.page_70_0": "1. 游戏玩法",
  "book.casinocraft.head_70_1": "游戏玩法",
  "book.casinocraft.page_70_1": "按回车键插入一个代币。如果激活了投注，最多可以放置5个代币。每个代币会点亮一个灯，允许你从新的方向连接图标。按空格键旋转轮盘。再次按空格键停止每个轮盘。当相同的3个图标在一个方向上对齐时，你就赢了。"
}
```
When I enter the game, everything is fine, but when I open the guidebook, it causes a memory leak and exits the game. I've checked the format of the book's content and it seems like there are no fatal errors. Can you help me with this?

**Comment by QianFuv** [comment]
error log:
```
[036月2024 00:07:22.997] [Render thread/ERROR] [net.minecraft.client.Minecraft/FATAL]: Out of memory
java.lang.OutOfMemoryError: Java heap space
	at mod.lucky77.screen.ScreenBook.createTextField(ScreenBook.java:209) ~[Lucky77_1.19.2_v04.jar%23154!/:1.0]
	at mod.lucky77.screen.ScreenBook.loadPages(ScreenBook.java:159) ~[Lucky77_1.19.2_v04.jar%23154!/:1.0]
	at mod.lucky77.screen.ScreenBook.<init>(ScreenBook.java:40) ~[Lucky77_1.19.2_v04.jar%23154!/:1.0]
	at mod.lucky77.system.SystemPlayer.openBookScreen(SystemPlayer.java:19) ~[Lucky77_1.19.2_v04.jar%23154!/:1.0]
	at mod.lucky77.item.ItemBook.m_7203_(ItemBook.java:50) ~[Lucky77_1.19.2_v04.jar%23154!/:1.0]
	at net.minecraft.world.item.ItemStack.m_41682_(ItemStack.java:266) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.multiplayer.MultiPlayerGameMode.m_233716_(MultiPlayerGameMode.java:384) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.multiplayer.MultiPlayerGameMode$$Lambda/0x000001a48e137b60.m_233885_(Unknown Source) ~[?:?]
	at net.minecraft.client.multiplayer.MultiPlayerGameMode.m_233729_(MultiPlayerGameMode.java:252) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.multiplayer.MultiPlayerGameMode.m_233721_(MultiPlayerGameMode.java:372) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.Minecraft.m_91277_(Minecraft.java:1693) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.Minecraft.m_91279_(Minecraft.java:1953) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.Minecraft.m_91398_(Minecraft.java:1768) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1078) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:700) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.main.Main.m_239872_(Main.java:212) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[client-1.19.2-20220805.130853-srg.jar%23155!/:?]
	at java.lang.invoke.LambdaForm$DMH/0x000001a48d005400.invokeStatic(LambdaForm$DMH) ~[?:?]
	at java.lang.invoke.LambdaForm$MH/0x000001a48d006800.invoke(LambdaForm$MH) ~[?:?]
	at java.lang.invoke.Invokers$Holder.invokeExact_MT(Invokers$Holder) ~[?:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invokeImpl(DirectMethodHandleAccessor.java:154) ~[?:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:27) ~[fmlloader-1.19.2-43.3.13.jar%23150!/:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler$$Lambda/0x000001a48d3bd688.run(Unknown Source) ~[?:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.8.jar%23137!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.8.jar%23137!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.8.jar%23137!/:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) ~[modlauncher-10.0.8.jar%23137!/:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) ~[modlauncher-10.0.8.jar%23137!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.8.jar%23137!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.8.jar%23137!/:?]
```

**Comment by QianFuv** [comment]
@Epidra

**Comment by Epidra** [comment]
Sorry, Windows Mail considers Github messages as spam, so I got the notification a little late.

I looked through everything and the problem comes from chinese not using spaces. The mod needs them to know how to divide the text into several lines for the book. Without spaces the divider never ends which causes a mermory overflow.

I added spaces after the periods, which lets me open the book:
![Screenshot (93)](https://github.com/Epidra/CasinoCraft/assets/17004012/a9259dc9-bf6d-4189-b7bb-cbc3e18cc328)
I can update the mod to recognize chinese punctuation marks, but in this case it would be better if you add "/b" after every few words.

/b forces a new line, which would look like this then: (I set the /b randomly)
![Screenshot (94)](https://github.com/Epidra/CasinoCraft/assets/17004012/a8fafee2-0666-45b3-b58b-b10ff3918c58)
